### PR TITLE
Use %w instead of %v to format errors

### PR DESCRIPTION
**Pull request by tanjing2020**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Ref #94437

#### Special notes for your reviewer:
@alculquicondor 

#### Does this PR introduce a user-facing change?

```
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by chendave — COMMENTED** [review]
I guess the commit subject need to be updated, iiuc, this is not wrapping anything, and most of case "%w" is identical to "%v" but seems the "%w" is better under some situation.

**Review by gavinfish — COMMENTED** [review]
`%w` is never better than `%v`, and the difference is that whether we want to expose the underlying errors to callers. So please check one by one instead of search and replace all.

**Comment by alculquicondor** [comment]
> `%w` is never better than `%v`

Why so? There are certain scenarios where we need to compare underlying errors. Just migrating everything to wrapped errors keeps consistency, so that we are ready to check errors when needed.

File: pkg/scheduler/framework/plugins/noderesources/fit.go [hunk 1]
```diff
@@ -179,7 +179,7 @@ func getPreFilterState(cycleState *framework.CycleState) (*preFilterState, error
 	c, err := cycleState.Read(preFilterStateKey)
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/noderesources/fit.go:179]: It looks like something changed the permissions of this file. Please revert.

**Review by alculquicondor — COMMENTED** [review]
To keep this PR focused, limit the changes to pkg/scheduler/framework, as those are the errors we are more likely to care about being wrapped.

**Comment by gavinfish** [comment]
> > `%w` is never better than `%v`
> 
> Why so? There are certain scenarios where we need to compare underlying errors. Just migrating everything to wrapped errors keeps consistency, so that we are ready to check errors when needed.

aha, what I want to say is `%w is not always better than %v` indeed. Yes, we should expose some underlying errors case by case.

**Comment by alculquicondor** [comment]
I don't think picking is worth the effort, given that the downsides of migrating everything are negligible.

File: pkg/scheduler/framework/plugins/noderesources/fit.go (hunk 1 above)
Comment (tanjing2020) [reply pkg/scheduler/framework/plugins/noderesources/fit.go:179]: Done.

File: pkg/scheduler/framework/plugins/serviceaffinity/service_affinity.go [hunk 2]
```diff
@@ -108,14 +108,14 @@ func (pl *ServiceAffinity) createPreFilterState(pod *v1.Pod) (*preFilterState, e
 	// Store services which match the pod.
 	matchingPodServices, err := helper.GetPodServices(pl.serviceLister, pod)
 	if err != nil {
-		return nil, fmt.Errorf("listing pod services: %v", err.Error())
+		return nil, fmt.Errorf("listing pod services: %w", err.Error())
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/serviceaffinity/service_affinity.go:111]: oh, don't call `err.Error()`, because that's a string

Comment (tanjing2020) [reply pkg/scheduler/framework/plugins/serviceaffinity/service_affinity.go]: I was too careless to notice the problem. Thank you very much. It has been revised.

**Comment by alculquicondor** [comment]
you have to fix the tests

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go [hunk 3]
```diff
@@ -18,11 +18,13 @@ package interpodaffinity
...
 import (
 	"context"
+	"errors"
+	"fmt"
 	"reflect"
 	"strings"
 	"testing"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (gavinfish) [inline pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:27]: nit: this change is unnecessary.

Comment (tanjing2020) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go]: Done.

File: pkg/scheduler/framework/plugins/serviceaffinity/service_affinity_test.go [hunk 4]
```diff
@@ -618,7 +619,7 @@ func TestPreFilterDisabled(t *testing.T) {
 	}
 	cycleState := framework.NewCycleState()
 	gotStatus := p.Filter(context.Background(), cycleState, pod, nodeInfo)
-	wantStatus := framework.AsStatus(fmt.Errorf(`error reading "PreFilterServiceAffinity" from cycleState: not found`))
+	wantStatus := framework.AsStatus(fmt.Errorf(`error reading "PreFilterServiceAffinity" from cycleState: %w`, errors.New("not found")))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/serviceaffinity/service_affinity_test.go:622]: Instead, export the error in `framework/cycle_state.go`? That is the benefit of wrapping, that we can compare the exact error. Now, we shouldn't be using reflect, but that's a separate problem.

File: pkg/scheduler/framework/plugins/noderesources/fit_test.go [hunk 5]
```diff
@@ -437,7 +438,7 @@ func TestPreFilterDisabled(t *testing.T) {
 	}
 	cycleState := framework.NewCycleState()
 	gotStatus := p.(framework.FilterPlugin).Filter(context.Background(), cycleState, pod, nodeInfo)
-	wantStatus := framework.NewStatus(framework.Error, `error reading "PreFilterNodeResourcesFit" from cycleState: not found`)
+	wantStatus := framework.AsStatus(fmt.Errorf(`error reading "PreFilterNodeResourcesFit" from cycleState: %w`, errors.New(framework.NotFound)))
```
Comment (alculquicondor) [inline pkg/scheduler/framework/plugins/noderesources/fit_test.go:441]: can you make `framework.NotFound` an error instead of a string?

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**